sdk/core/azure_core/src/hmac.rs

```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-use crate::auth::Secret;
+use crate::authentication::Secret;
```

Comment:
@JeffreyRichter what do you think about this? @LarryOsterman and I were talking and perhaps we should just go back to "auth", but I think all those years of .NET development and an anti-abbreviation stance worse on me. That said, Azure SDK for C++ uses namespace `Azure::Core::Credentials` so we could use "credentials" instead since everything therein support credentials.

*Naming is hard example 2,349,208*

Reply:
/cc @analogrelay

Reply:
At least with "auth", it's "deliciously ambiguous", as @LarryOsterman put it, in that it can mean "authentication" or "authorization". Still, if we want to avoid abbreviations I think "credentials" accomplishes the same.

Reply:
I'm generally anti-abbreviation unless absolutely necessary, but it's not an immovable stance ;).

> At least with "auth", it's "deliciously ambiguous", as @LarryOsterman put it, in that it can mean "authentication" or "authorization". Still, if we want to avoid abbreviations I think "credentials" accomplishes the same.

Is there any authorization going on here? I'd presume not since this is a client. Authorization is the server's responsibility and if we have APIs to manage AuthZ in the future they would live in a separate module anyway.

If we do anticipate a need to handle authorization components here, then I'd lean towards a more general (if still ambiguous) term like "Security" rather than **intentionally** create ambiguity between two specific concepts.

Reply:
I like authentication. I agree with Ashley on being anti-abbreviation (except if I'm writing Go).

Reply:
Technically, a service authenticates and authorizes (often separate services) so a client really does neither. We can't just say who we are with any authority. 😉 And the HTTP header is "Authorization" in requests or "WWW-Authenticate" in responses, and in with some policies we use both.

I'm starting to lean toward "credentials" like C++ if we agree on no abbreviations. We're idiomatic, but I still believe in consistency across languages if there's no other reason not to: it helps foster familiarity for polyglot projects.

Reply:
The crate defines credential types, right? If so, credentials seems the best to me.

Reply:
We're talking about a submodule of `azure_core` but, yes, it defines credential-*associated* types including `trait TokenCredential`, `struct AccessToken`, etc.

Reply:
> Technically, a service authenticates and authorizes (often separate services) so a client really does neither. We can't just say who we are with any authority. 😉

![image](https://github.com/user-attachments/assets/b8c52b3c-c10c-437f-b41f-e27d2683e23a)

> The crate defines credential types, right? If so, credentials seems the best to me.

I like this. `credentials` feels right to me.

Reply:
Renamed to `azure_core::credentials`.